### ISSUE TITLE
Just 4 local processes are used, despite setting DISTCC_HOSTS, /etc/distcc/hosts, ~/.distcc/hosts

### ISSUE THREAD

**Issue by JStech** [distcc/distcc#396]
I'm using distcc to build [PCL](https://pointclouds.org), and I've been unable to get it to work reliably with my server. It did work once, but I haven't been able to recreate it. Typically it just runs four processes on localhost (my laptop), although sometimes it will briefly send one or two jobs to the server at the beginning of the build. Making me even more suspicious, `distccmon-gnome` and `distccmon-text` show different states--I see four jobs running on localhost in the text monitor, and nothing in the GUI.

Here are the relevant environment variables on the laptop:

    DISTCC_HOSTS=localhost/6 192.168.1.254/24
    DISTCC_DIR=/tmp/distcc
    CC=distcc /usr/bin/x86_64-linux-gnu-gcc-7
    CXX=distcc /usr/bin/x86_64-linux-gnu-g++-7
    PATH=/home/john/gems/bin:/usr/lib/ccache:/home/john/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
    DISTCC_VERBOSE=1

I've also put `localhost/6 192.168.1.254/24` in `/etc/distcc/hosts`, `/tmp/distcc/hosts`, and `~/.distcc/hosts`.

To build, I first run `cmake`, and I confirmed that the `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` variables are correctly set to `distcc`, with the absolute path to the actual compiler as `*_ARG1`. (I have ccache installed, so I'm using the absolute path to avoid using it to try to isolate this issue.) Then I run `make -j 30`.

Typically, `distccmon-gnome` will remain blank or show the one or two jobs sent to the server, while `distccmon-text` will show something like this:

     14442  Compile                                                   localhost[0]
     14333  Compile                                                   localhost[1]
     14332  Compile                                                   localhost[2]
     14329  Compile                                                   localhost[3]

Also, I've verified that the daemon is running on the server, through `nc -zv 192.168.1.254 3632`, and I can use the server when building ROS packages using `catkin`.

Like I said, I did get it to work once--there were lots of jobs running in `distccmon-gnome`, and the whole library built in a few minutes. When it happened, I wasn't sure what I'd done to make it work, and then it stopped working after I'd walked away and came back. I'd appreciate any help.

**Comment by shawnl** [comment]
> distccmon-gnome and distccmon-text show different states--I see four jobs running on localhost in the text monitor, and nothing in the GUI.

Local compiles are very different from distributed compiles, so if that is the only difference I wouldn't be worried about it.

> CC=distcc /usr/bin/x86_64-linux-gnu-gcc-7
> CXX=distcc /usr/bin/x86_64-linux-gnu-g++-7

I think this is your problem. If you use absolute paths you need to pass `--enable-tcp-insecure` when you start `distccd`. Please read the man pages for the details on this security feature. You can add /usr/lib/distcc to your `PATH`, and then just call `distcc` as if it was your compiler through a symlink there.

**Comment by JStech** [comment]
Thanks for your help. Unfortunately that didn't fix it. I restarted the daemon with `--enable-tcp-insecure` and still just got four local jobs, so I also re-ran cmake with `CC="distcc gcc-7" CXX="distcc g++-7" cmake ...`, but still no luck.

So distccmon-gnome and distccmon-text are not just different ways to monitor the same thing? Can you explain more about the difference between "local" and "distributed" compiles, from distcc's perspective? It sounds like I'm wanting a distributed compile but distcc is running a local compile.

**Comment by shawnl** [comment]
>  It sounds like I'm wanting a distributed compile but distcc is running a local compile.

distcc will tell you this, if you run it as you should be with `DISTCC_VERBOSE=1`. there is also `DISTCC_FALLBACK=0` to refuse to compile locally. You should also use logging with distccd, with is a little trickier to use (see the man page).

distccmon-gnome and distccmon-text use the same protocol. I am just saying that the way local compiles are shown would not be a source of worry. There is some documentation of that protocol in the source tree, but for me those tools have always just worked, and this is probably not your problem.

**Comment by JStech** [comment]
Got it figured out. I looked more closely at the verbose output, and there was a compiler flag that was forcing it to build locally. I was able to change the PCL build options to disable the flag (`-DPCL_ENABLE_SSE:BOOL=FALSE`), and it's distributing again. I'm not sure why I was able to get it to distribute the one time before, but I've been switching between branches a lot, so it might've just been a branch that didn't have that flag set to begin with.

Thanks for your help.
